packages/visual-editor/src/utils/getLocationPath.ts

```diff
@@ -20,10 +22,12 @@ export const getLocationPath = (
     return `${relativePrefixToRoot ?? ""}${location.slug}`;
   }
 
-  const localePath = locale !== "en" ? `${locale}/` : "";
+  const isPrimaryLocale = location.__?.isPrimaryLocale ?? false;
+
+  const localePath = isPrimaryLocale ? "" : `${locale ?? location.locale}/`;
   const path = location.address
     ? `${localePath}${location.address.region}/${location.address.city}/${location.address.line1}-${location.id}`
```

Comment:
If this is only used for nearby/last level DM page we'll need to align with what we backfill to, which won't have the suffix.